### DISCUSSION TITLE
Start background task independent of connection on Heroku

### DISCUSSION THREAD

**Discussion by pdneumann** [miguelgrinberg/Flask-SocketIO#1820] [Q&A] [answered]
I'm running into a stumbling block while trying to deploy an app to Heroku.  I've reproduced it using the app.py script in the example.  I've moved the start_background_task call as you suggested here [https://stackoverflow.com/questions/53379228/flask-socketio-how-to-share-data-between-python-thread-and-socketio-start-backg](https://stackoverflow.com/questions/53379228/flask-socketio-how-to-share-data-between-python-thread-and-socketio-start-backg).  I commented out start_background_task call in the connect() function so I have it as below:

    if __name__ == '__main__':
        socketio.start_background_task(background_thread)
        socketio.run(app)

This works when deployed locally but not when deployed to Heroku.  Locally, it emits and the count continues up regardless of connection as expected.  On Heroku, the connection message emits but the server event never fires.

I'm using a runtime.txt file to use Python 3.8:
`python-3.8.13`

My requirements.txt file is as follows:

    bidict==0.22.0
    cffi==1.15.0
    click==8.1.2
    colorama==0.4.4
    Flask==2.1.1
    Flask-SocketIO==5.1.2
    gevent==21.12.0
    gevent-websocket==0.10.1
    greenlet==1.1.2
    gunicorn==20.1.0
    importlib-metadata==4.11.3
    itsdangerous==2.1.2
    Jinja2==3.1.1
    MarkupSafe==2.1.1
    pycparser==2.21
    python-engineio==4.3.2
    python-socketio==5.6.0
    typing-extensions==4.2.0
    Werkzeug==2.1.1
    zipp==3.8.0
    zope.event==4.5.0
    zope.interface==5.4.0

And finally my Procfile is this:
`web: gunicorn -k geventwebsocket.gunicorn.workers.GeventWebSocketWorker -w 1 app:app`

I'm probably missing something, but cannot seem to figure out what.  I have a couple of threads that I want to start immediately rather than wait for connections to fire them.

**Comment by miguelgrinberg** [comment, marked as answer]
It does not work because you are running Gunicorn. The code in the `if __name__ == '__main__'` conditional never executes when using Gunicorn. The easiest way to make this work is to run on the Gevent web server by running `python app.py`, like you are doing locally.

**Reply by pdneumann** [reply]
Wow.  That is a serious facepalm moment.  Thank you for the advice and quick reply!
